volmdlr/faces.py

```diff
@@ -6466,78 +6466,86 @@ def rotation_inplace(self, center: volmdlr.Point3D, axis: volmdlr.Vector3D,
         self.bounding_box = new_bounding_box
 
     def subdescription(self, resolution=0.01):
-        frame = self.surface3d.frame
-        pts2d = [pt.to_2d(frame.origin, frame.u, frame.v) for pt in self.points]
-
-        t_poly2d = volmdlr.wires.ClosedPolygon2D(pts2d)
-
-        xmin, xmax = min(pt.x for pt in pts2d), max(pt.x for pt in pts2d)
-        ymin, ymax = min(pt.y for pt in pts2d), max(pt.y for pt in pts2d)
-
-        nbx, nby = int(((xmax - xmin) / resolution) + 2), int(((ymax - ymin) / resolution) + 2)
-        points_box = []
-        for i in range(nbx):
-            x = min(xmin + i * resolution, xmax)
-            if x == xmin:
-                x = xmin + 0.01 * resolution
-            for j in range(nby):
-                y = min(ymin + j * resolution, ymax)
-                if y == ymin:
-                    y = ymin + 0.01 * resolution
-                points_box.append(volmdlr.Point2D(x, y))
-
-        points = [pt.copy() for pt in self.points]
-        for pt in points_box:
-            if t_poly2d.point_belongs(pt):
-                points.append(pt.to_3d(frame.origin, frame.u, frame.v))
-            elif t_poly2d.point_over_contour(pt):
-                points.append(pt.to_3d(frame.origin, frame.u, frame.v))
-
-        return volmdlr.Vector3D.remove_duplicate(points)
+
+        lengths = [self.points[0].point_distance(self.points[1]),
+                   self.points[1].point_distance(self.points[2]),
+                   self.points[2].point_distance(self.points[0])]
+        ls_max = max(lengths)
+
+        if ls_max <= resolution:
+            return self.points
+
+        pos_ls_max = lengths.index(ls_max)
+        p0, p1, p2 = self.points[-3 + pos_ls_max], self.points[-3 + pos_ls_max + 1], self.points[-3 + pos_ls_max + 2]
+
+        v_init = p0 - p1
+        v_init.normalize()
+        p0_res = []
+
+        nb = int(ls_max / resolution) + 2
+        for k in range(nb):
+            if k == 0:
+                p0_res.append(p1)
+            l = min(k * resolution, ls_max)
+            p0_res.append(p1 + v_init * l)
+
+        v1, l1 = p2 - p1, p2.point_distance(p1)
+        v1.normalize()
+        p1_res = []
+
+        for k, p0_k in enumerate(p0_res):
+            if k == 0:
+                point_v1 = p1
+            l = min(p0_res[0].point_distance(p0_k) * l1 / ls_max, l1)
+            point_v1 = p1 + v1 * l
+
+            l_int = point_v1.point_distance(p0_k)
+            nb_int = int(l_int / resolution) + 2
+            if nb_int == 2:
+                p1_res.append(point_v1)
+            else:
+                v_int = point_v1 - p0_k
+                v_int.normalize()
+                step_in = l_int / (nb_int - 1)
+                for i in range(nb_int):
+                    l = min(i * step_in, l_int)
+                    if l != 0:
+                        p1_res.append(p0_k + v_int * l)
+
+        return npy.unique(p0_res + p1_res).tolist()
```

Comment:
Could you please try renaming some variables while being more explicit in what each variable is?

Reply:
I tried to change, can you confirm ?